docs/spec/sources.md

```diff
@@ -0,0 +1,342 @@
+critical to developing a distributed system that reacts to events.
+
+A Source:
+
+- Represents an off or on-cluster system, service or application that produces
+  events to be consumed by a sink.
+- Produces or imports CloudEvents.
+- Sends CloudEvents to the configured **sink**.
+
+In practice, sources are an abstract concept that allow us to create declarative
+configurations through the usage of Custom Resource Definitions (CRDs) extending
+Kubernetes. Those CRDs are instantiated by creating an instance of the resource.
+It is up to the implementation of the source author to understand the best way
+to realize the source application. This could be as 1:1 deployments inside of
+Kubernetes per resource, as a single multi-tenant application, or even an
+off-cluster implementation; or all combinations in-between.
+
+For Kubernetes Operators, there are two states to sources:
+
+1. A source CRD and controller (if required) have been installed into the
+   cluster.
+
+   A cluster with a source CRD installed allows the other operators in the
+   cluster to find and discover what is possible to source events from. These
+   CRDs define the credentials, input parameters, k8s categories, event types,
+   duck types, and aggregated RBAC roles required to interpret the source as the
+   source author expects. This topic is expanded upon in the
+   [Source CRDs](#source-crds) section.
+
+1. A source custom object has been created in the cluster.
+
+   Once a cluster operator creates an instance of a source and provides the
+   necessary credentials, input parameters, and event sink, the source
+   controller will realize this into whatever is required for that source for
+   the current situation. While this resource is running, a cluster operator
+   would like to inspect the resource without needing to be fully aware of the
+   implementation. This is done by conforming to the [Source]() ducktype. This
+   topic is expanded upon in the [Source Custom Objects](#source-custom-objects)
+   section.
+
+The goal of requiring CRD labels and running resource shapes is to aid in
+discovery and understanding of potential sources that could be leveraged in the
+cluster, and sources that exist and are running in the cluster.
+
+## Source CRDs
+
+Sources are more useful if they are discoverable. Knative Sources MUST use a
+ducktype label to allow controllers and operators the ability to find which CRDs
+are considered to be adhering to the
+[Source](https://godoc.org/github.com/knative/pkg/apis/duck/v1#Source) ducktype.
+
+CRDs that are to be understood as a `source` MUST be labeled:
+
+```yaml
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+  labels:
+    duck.knative.dev/source: "true" # <-- required to be a source.
+```
+
+Labeling sources in this way allows for cluster operators to list CRDs and
+filter on which ones are choosing to adhere to the source ducktype. To make this
```

Comment:
I think this would be clearer to say "allows developers to list sources by filtering the list of CRDs:"